registration/include/pcl/registration/impl/ndt.hpp

```diff
@@ -514,7 +691,7 @@ NormalDistributionsTransform<PointSource, PointTarget>::trialValueSelectionMT (d
     return a_s;
   }
   // Case 3 in Trial Value Selection [More, Thuente 1994]
-  if (std::fabs (g_t) <= std::fabs (g_l))
+  if (std::fabs (g_t) <= std::fabs (g_l) && a_t != a_l)
```

Comment:
This (and related) change seems to belong to a separate PR.

Reply:
OK, I'll open another PR. Note that this change must be merged before this NDT_OMP PR because the multi-threaded NDT can cause segfault without this additional condition check.